include/aws/common/error.h

```diff
@@ -0,0 +1,104 @@
+#ifndef AWS_COMMON_ERROR_H_
+
+/*
+* Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License").
+* You may not use this file except in compliance with the License.
+* A copy of the License is located at
+*
+*  http://aws.amazon.com/apache2.0
+*
+* or in the "license" file accompanying this file. This file is distributed
+* on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+* express or implied. See the License for the specific language governing
+* permissions and limitations under the License.
+*/
+
+#define AWS_COMMON_ERROR_H_
+
+#include <aws/common/exports.h>
+#include <stdint.h>
+
+struct aws_error_info {
+    int error_code;
+    const char *literal_name;
+    const char *error_str;
+    const char *lib_name;
+    const char *formatted_name;
+};
+
+struct aws_error_info_list {
```

Comment:
Would suggest having sizeof(aws_error_info) and sizeof(aws_error_info_list) fields in this struct for future expansion.

Reply:
lets come back to this one. We have some structs will have to do this for soon and I'd like to nail the pattern in that use case.